compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3577,6 +3577,128 @@ TR::Register *OMR::Power::TreeEvaluator::vsqrtEvaluator(TR::Node *node, TR::Code
      }
    }
 
+TR::Register *OMR::Power::TreeEvaluator::vminEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   switch(node->getDataType().getVectorElementType())
+     {
+     case TR::Int8:
+       return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vminsb);
+     case TR::Int16:
+       return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vminsh);
+     case TR::Int32:
+       return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vminsw);
+     case TR::Int64:
+       return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vminsd);
+     case TR::Float:
+       return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, TR::InstOpCode::vminfp);
+     case TR::Double:
+       return TR::TreeEvaluator::vminDoubleHelper(node, cg);
+     default:
+       TR_ASSERT(false, "unrecognized vector type %s\n", node->getDataType().toString()); return NULL;
+     }
+   }
+
+TR::Register *OMR::Power::TreeEvaluator::vminDoubleHelper(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
Do you assume this function can be called from outside of the `TreeEvaluator` class?
If not, how about making it a static function in this file instead of a member function of `TreeEvaluator`?

Reply:
Good point! This function will only ever be called from `vminEvaluator()`, so no need to make it a member of `TreeEvaluator`.